invenio_rdm_records/services/communities/service.py

```diff
@@ -355,3 +358,49 @@ def set_default(self, identity, id_, data, uow):
         )
 
         return record.parent
+
+    @unit_of_work()
+    def bulk_add(self, identity, id_, recids, set_default=False, uow=None):
+        """Bulk adds records to a community.
+
+        :param identity: The identity performing the action.
+        :param id_: The ID of the community.
+        :param recids: List of record IDs to be added to the community.
+        :param set_default:  Whether to set the community as default for the added records.
+        """
+        self.require_permission(identity, "bulk_add")
+        for record_id in recids:
+            record = self.record_cls.pid.resolve(record_id)
+            community = current_communities.service.read(identity, id_)
+            self._include_one(record, community._record, set_default=set_default)
+            # Commit and re-index everything
+            uow.register(
+                ParentRecordCommitOp(
+                    record.parent,
+                    indexer_context=dict(service=current_rdm_records_service),
+                )
+            )
+            uow.register(
+                RecordCommitOp(
+                    record,
+                    indexer=current_rdm_records_service.indexer,
+                    index_refresh=True,
+                )
+            )
+
+    def _include_one(self, record, community, set_default=False):
+        """This service call differs from the `add` method in that it does not create a request to include the record in the community.
+
+        It adds the record to the community directly.
+        It should only be called from within the service.
+        """
+        default = set_default or not record.parent.communities
+
+        parent_community = getattr(community, "parent", None)
+        already_in_parent = parent_community and str(parent_community.id) in record
+
+        if parent_community and not already_in_parent:
+            record.parent.communities.add(parent_community, request=self.request)
+
+        record.parent.communities.add(community, request=None, default=default)
```

Comment:
Question: Do we have the use case that the community is already part of the record? (might be worth to add the same check as above)

Reply:
Good one, I will add a check and tests. Thanks!